taskchampion/xtask/src/main.rs

```diff
@@ -29,3 +54,110 @@ fn codegen() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+/// `cargo xtask msrv (X.Y)`
+///
+/// This checks and updates the Minimum Supported Rust Version for all files specified in the tuple list `MSRV_PATH_REGEX` const in `xtask/src/main.rs` where the regex pattern specified is found in the file.
+/// The argument "X.Y" will replace existing X.Y any place the regex pattern specified in the tuple list `MSRV_PATH_REGEX` is found.
+fn msrv(args: Vec<String>) -> anyhow::Result<()> {
+    if args.len() != 2 {
+        // check that (X.Y) argument is (mostly) valid:
+        if !args[2].chars().all(|c| c.is_numeric() || c == '.') {
+            anyhow::bail!("xtask: Invalid argument format. Xtask MSRV argument takes the form \"X.Y(y)\", where XYy are numbers. eg: `cargo run xtask MSRV 1.68`");
+        }
+
+        // set regex for replacing version number only within the pattern found within a line
+        let re_msrv_version = Regex::new(r"([0-9]+(\.|[0-9]+|))+").unwrap();
+
+        // for each file in const paths vector
+        for msrv_file in MSRV_PATH_REGEX {
+            let path: &Path = Path::new(&msrv_file.0);
+            let mut is_pattern_in_file = false;
+
+            // check if path exists, if not, skip then continue
+            if !path.exists() {
+                println!(
+                    "xtask: Warning: xtask: Path \"{}\" not found. Skipping.",
+                    path.display()
+                );
+                continue;
+            }
+
+            let mut file: File = File::options().read(true).write(true).open(path)?;
+            let reader = BufReader::new(&file);
+
+            // set search string and the replacement string for version number content
+            let re_msrv_pattern = Regex::new(msrv_file.1).unwrap();
+            let version_replacement_string = &args[2];
+
+            // for each line in file
+            let mut file_string = String::new();
+            for line in reader.lines() {
+                let line_ref = &line.as_ref().unwrap();
```

Comment:
The auto-deref functionality of the language will automatically make references to an object when calling a method on it, so this is the same as:
```suggestion
                let line_ref = &line.unwrap();
```